featuretools/primitives/rolling_primitive_utils.py

```diff
@@ -153,24 +140,104 @@ def _check_window_size(window_size):
         raise TypeError("Window length must be either an offset string or an integer.")
 
 
-def _check_gap(window_size, gap):
+def _check_gap(window_size: Union[int, str], gap: Union[int, str]) -> None:
     # Gap must either be a valid offset string that also has an offset string window length
     if isinstance(gap, str):
         if not isinstance(window_size, str):
             raise TypeError(
                 f"Cannot roll series with offset gap, {gap}, and numeric window length, {window_size}. "
                 "If an offset alias is used for gap, the window length must also be defined as an offset alias. "
-                "Please either change gap to be numeric or change window length to be an offset alias."
+                "Please either change gap to be numeric or change window length to be an offset alias.",
             )
         try:
             to_offset(gap)
         except ValueError:
             raise ValueError(
-                f"Cannot roll series. The specified gap, {gap}, is not a valid offset alias."
+                f"Cannot roll series. The specified gap, {gap}, is not a valid offset alias.",
             )
     # Or an integer greater than or equal to zero
     elif isinstance(gap, int):
         if gap < 0:
             raise ValueError("Gap must be greater than or equal to zero.")
     else:
         raise TypeError("Gap must be either an offset string or an integer.")
+
+
+def apply_rolling_agg_to_series(
+    series: Series,
+    agg_func: Callable[[Series], float],
+    window_size: Union[int, str],
+    gap: Union[int, str] = 0,
+    min_periods: int = 1,
+    ignore_window_nans: bool = False,
+) -> np.ndarray:
+    """Applies a given aggregation function to a rolled series.
+
+    Args:
+        series (Series): The series over which rolling windows will be created. Must be numeric in nature
+            and have a DatetimeIndex.
+        agg_func (callable[Series -> float]): The aggregation function to apply to a rolled series.
+        window_size (int, string): Specifies the amount of data included in each window.
+            If an integer is provided, will correspond to a number of rows. For data with a uniform sampling frequency,
+            for example of one day, the window_length will correspond to a period of time, in this case,
+            7 days for a window_length of 7.
+            If a string is provided, it must be one of pandas' offset alias strings ('1D', '1H', etc),
+            and it will indicate a length of time that each window should span.
+            The list of available offset aliases, can be found at
+            https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases
+        gap (int, string, optional): Specifies a gap backwards from each instance before the
+            window of usable data begins. If an integer is provided, will correspond to a number of rows.
```

Comment:
```suggestion
            window of usable data begins. If an integer is provided, will correspond to the number of rows.
```

Reply:
First one actually makes more sense in the context of how gap works.